### PR TITLE
fix tests

### PR DESCRIPTION
**Pull request by almogdepaz**
### Testing Notes:

---

> [!NOTE]
> **Low Risk**
> Test-only changes that primarily adjust assertions and fixtures for consensus-mode edge cases; low production risk but may hide real regressions if expectations are now too permissive.
> 
> **Overview**
> Improves test robustness across consensus modes by removing/relaxing some `limit_consensus_modes` gates and updating expectations to match HF3.0 chain selection rules (e.g., treating equal-weight reorg candidates as `NEW_PEAK` only when `total_iters` wins, otherwise `ADDED_AS_ORPHAN`).
> 
> Stabilizes specific flaky cases: `test_compact_protocol_invalid_messages` now derives the “wrong” VDF from a field less likely to collide when iterations are small; timelord new-peak orphaning test avoids overflow via `skip_overflow=True`; and several tests add clarifying HF3.0/v2-plot-related notes and adjust assumptions about pool-target/plot outcomes.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 5c67525bce9551810f710a26046e93cf18c30c7d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: chia/_tests/blockchain/test_blockchain.py [hunk 1]
```diff
@@ -3480,22 +3472,23 @@ async def test_long_reorg(
                 await _validate_and_add_block(
                     b, reorg_block, expected_result=AddBlockResult.ALREADY_HAVE_BLOCK, augmented_blockchain=aug_chain
                 )
-            elif reorg_block.weight <= chain_1_weight:
+            else:
                 if fork_info2 is None:
                     fork_info2 = ForkInfo(reorg_block.height - 1, reorg_block.height - 1, reorg_block.prev_header_hash)
+                peak = b.get_peak()
+                assert peak is not None
+                if reorg_block.weight > peak.weight or (
```
Comment (arvidn) [inline chia/_tests/blockchain/test_blockchain.py]: it makes me a bit nervous to have this be conditional. If something is changed with the chains, this test may not trigger a reorg anymore, but it will still pass. At least that's what it looks like.

I'm thinking that maybe it would be possible to fix this by adjusting `chain_1_weight` instead, to depend on which chain we're using.

File: chia/_tests/blockchain/test_blockchain.py [hunk 2]
```diff
@@ -3847,9 +3833,11 @@ async def test_reorg_new_ref(empty_blockchain: Blockchain, bt: BlockTools) -> No
             # same height as peak decide by iterations
             peak = b.get_peak()
             assert peak is not None
-            # same height as peak should be ADDED_AS_ORPHAN if  block.total_iters >= peak.total_iters
-            assert block.total_iters < peak.total_iters
-            expected = AddBlockResult.NEW_PEAK
+            # same height as peak should be ADDED_AS_ORPHAN if block.total_iters >= peak.total_iters
+            if block.total_iters < peak.total_iters:
```
Comment (arvidn) [inline chia/_tests/blockchain/test_blockchain.py:3846]: the reason this check is here is to make sure the test actually triggers a reorg. As far as I can tell, with this change, we wouldn't ensure that happens.

Would it not be possible to make the check against `19` above instead depend on which chain we're using, but still with a hard coded height.

File: chia/_tests/blockchain/test_blockchain.py (hunk 1 above)
Comment (almogdepaz) [reply chia/_tests/blockchain/test_blockchain.py]: there is already an assert that a reorg happens https://github.com/Chia-Network/chia-blockchain/pull/20457/changes#diff-72c7bdde0041eaa653e8a3ed2eb20c885f91b4a1ab37059e21c332b5f3dadd2fL3555

**Review by arvidn — APPROVED** [review]
